test/Datadog.Trace.ClrProfiler.IntegrationTests/Helpers/TestHelper.cs

```diff
@@ -169,21 +169,51 @@ public ProcessResult RunSampleAndWaitForExit(int traceAgentPort, int? statsdPort
         }
 
-        public (Process Process, string ConfigFile) StartIISExpress(int traceAgentPort, int iisPort, bool classicMode)
+        public (Process Process, string ConfigFile) StartIISExpress(int traceAgentPort, int iisPort, IisAppType appType)
         {
             // get full paths to integration definitions
             IEnumerable<string> integrationPaths = Directory.EnumerateFiles(".", "*integrations.json").Select(Path.GetFullPath);
 
-            var exe = EnvironmentHelper.GetSampleExecutionSource();
+            var iisExpress = EnvironmentHelper.GetIisExpressPath();
+
+            var appPool = appType switch
+            {
+                IisAppType.AspNetClassic => "Clr4ClassicAppPool",
+                IisAppType.AspNetIntegrated => "Clr4IntegratedAppPool",
+                IisAppType.AspNetCoreInProcess => "UnmanagedClassicAppPool",
+                IisAppType.AspNetCoreOutOfProcess => "UnmanagedClassicAppPool",
+                _ => throw new InvalidOperationException($"Unknown {nameof(IisAppType)} '{appType}'"),
+            };
+
+            var appPath = appType switch
+            {
+                IisAppType.AspNetClassic => EnvironmentHelper.GetSampleProjectDirectory(),
+                IisAppType.AspNetIntegrated => EnvironmentHelper.GetSampleProjectDirectory(),
+                IisAppType.AspNetCoreInProcess => EnvironmentHelper.GetSampleApplicationOutputDirectory(),
+                IisAppType.AspNetCoreOutOfProcess => EnvironmentHelper.GetSampleApplicationOutputDirectory(),
+                _ => throw new InvalidOperationException($"Unknown {nameof(IisAppType)} '{appType}'"),
+            };
 
             var configTemplate = File.ReadAllText("applicationHost.config");
 
             var newConfig = Path.GetTempFileName();
 
             configTemplate = configTemplate
-                            .Replace("[PATH]", EnvironmentHelper.GetSampleProjectDirectory())
+                            .Replace("[PATH]", appPath)
                             .Replace("[PORT]", iisPort.ToString())
-                            .Replace("[POOL]", classicMode ? "Clr4ClassicAppPool" : "Clr4IntegratedAppPool");
+                            .Replace("[POOL]", appPool);
+
+            var isAspNetCore = appType == IisAppType.AspNetCoreInProcess || appType == IisAppType.AspNetCoreOutOfProcess;
+            if (isAspNetCore)
+            {
+                var hostingModel = appType == IisAppType.AspNetCoreInProcess ? "inprocess" : "outofprocess";
+                configTemplate = configTemplate
+                                .Replace("[DOTNET]", EnvironmentHelper.GetDotnetExe())
+                                .Replace("[RELATIVE_SAMPLE_PATH]", $".\\{EnvironmentHelper.GetSampleApplicationFileName()}")
+                                .Replace("[HOSTING_MODEL]", hostingModel);
+            }
+
+            // If running a .NET Core app, need to set
```

Comment:
I'm not sure what this comment is referring to. Should this comment be moved up several lines?

Reply:
I think it's just a left over comment from an intermediate state tbh, I'll delete it